workload: validate bank rows flag

herkolategan opened the pull request:
Previously, if a value of 1 was passed in for rows, the workload would panic. We
need at least two accounts to be able to do transfers. This change ensures we
validate the flag.

All unit tests that pass 0 or 1 have been patched to pass at least 2 accounts.

Fixes: #153849

herkolategan committed:
workload: validate bank rows flag

Previously, if a value of 1 was passed in for rows, the workload would panic. We
need at least two accounts to be able to do transfers. This change ensures we
validate the flag.

Fixes: #153849

herkolategan committed:
backup: use more than 1 row in bank workload

The validation added in the bank workload now requires at least two rows for
validation to pass. This change updates the unit tests to pass at least 2
accounts.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/157050)
<!-- Reviewable:end -->

herkolategan committed:
bank: update the ops function to consider validation

Previously, the logic in the bank `Ops` function tried to work around the fact
that the number of rows (accounts) might be 0 or 1. This logic didn't seem
right, as the `to` bank account was never allowed to be on the tail end.

This change uses the condition enforced by the validation that we always expect
rows to be at least 2. This allows us to simplify the from and to selection
here, which is not allowed to point to the same account.

herkolategan commented on pkg/workload/bank/bank.go:
```diff
@@ -236,10 +240,11 @@ func (b *bank) Ops(
+			// Rows are always expected to be at least two (via validation).
 			from := rng.IntN(b.rows)
-			to := rng.IntN(b.rows - 1)
-			for from == to && b.rows != 1 {
```
> ## Potential Bug(s) Detected

If we have 2 rows (allowed), this could potentially indefinitely loop.
Both from and to select 0, hence `from == to` and `b.rows != 1 (= 2)`.
And `b.rows - 1 = 1` resulting in to staying 0 during the loop, resulting in an indefinite loop.

With a minimum of 2 rows now being validated we can update this logic to account for that.
